nexus/types/src/deployment/clickhouse.rs

```diff
@@ -0,0 +1,118 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Types used in blueprints related to clickhouse configuration
+
+use clickhouse_admin_types::{KeeperId, ServerId};
+use omicron_common::api::external::Generation;
+use omicron_uuid_kinds::OmicronZoneUuid;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use uuid::Uuid;
+
+/// Global configuration for all clickhouse servers (replicas) and keepers
+#[derive(Clone, Debug, Eq, PartialEq, JsonSchema, Deserialize, Serialize)]
+pub struct ClickhouseClusterConfig {
+    /// The last update to the clickhouse cluster configuration
+    ///
+    /// This is used by `clickhouse-admin` in the clickhouse server and keeper
+    /// zones to discard old configurations.
+    pub generation: Generation,
+    /// Clickhouse Server ids must be unique and are handed out monotonically.
+    /// Keep track of the last used one.
+    pub max_used_server_id: ServerId,
+    /// CLickhouse Keeper ids must be unique and are handed out monotonically.
+    /// Keep track of the last used one.
+    pub max_used_keeper_id: KeeperId,
+    /// An arbitrary name for the Clickhouse cluster shared by all nodes
+    pub cluster_name: String,
+    /// An arbitrary string shared by all nodes used at runtime to determine
+    /// whether nodes are part of the same cluster.
+    pub cluster_secret: String,
```

Comment:
At the moment we're hardcoding this value into the ClickHouse config templates. What would this field be used for?

https://github.com/oxidecomputer/omicron/blob/fef8616bef83cc21d16dc7951b3f48fa88a0b106/clickhouse-admin/types/src/config.rs#L209-L210

Should we wait to handle this value securely before adding it to this config struct? https://github.com/oxidecomputer/omicron/issues/3823

Reply:
I think the name 'secret' is a misnomer. It doesn't provide any real security guarantees. For those we'll end up using certificates on the control plane for authentication. The main usage is just to make sure that nodes run accidentally with access to each other's IPs aren't actually talked to. Quoting the [clickhouse replication docs](https://clickhouse.com/docs/en/architecture/replication):
> A secret is created for the cluster named cluster_1S_2R with the value mysecretphrase. The secret is shared across all of the remote servers in the environment to ensure that the correct servers are joined together.

So I think we'll end up just generating a UUID for this and stringifying it.

Reply:
Ahhh, yeah, ok that makes sense.